packages/surveys/src/components/general/LanguageSwitch.tsx

```diff
@@ -0,0 +1,85 @@
+import { useRef, useState } from "react";
+
+import { getLanguageLabel } from "@formbricks/lib/i18n/utils";
+import { useClickOutside } from "@formbricks/lib/utils/hooks/useClickOutside";
+import { TSurveyLanguage } from "@formbricks/types/surveys";
+
+interface LanguageSwitchProps {
+  selectedLanguageCode: string;
+  surveyLanguages: TSurveyLanguage[];
+  setSelectedLanguageCode: (languageCode: string) => void;
+  setFirstRender?: (firstRender: boolean) => void;
+}
+export const LanguageSwitch = ({
+  surveyLanguages,
+  selectedLanguageCode,
+  setSelectedLanguageCode,
+  setFirstRender,
+}: LanguageSwitchProps) => {
+  const [showLanguageDropdown, setShowLanguageDropdown] = useState(false);
+  const toggleDropdown = () => setShowLanguageDropdown((prev) => !prev);
+  const languageDropdownRef = useRef(null);
+  const defaultLangauge = surveyLanguages.find((surveyLanguage) => {
+    return surveyLanguage.default === true;
+  })?.language;
+
+  const changeLanguage = (language: TSurveyLanguage) => {
```

Comment:
I think most of the time it's easier to work with language codes and don't pass the whole language object to reduce complexity. So I would also like to have the languageCode here in the params :-)